### PR TITLE
feat(gym): period-comparison stat row on Stair detail view (#77)

### LINKED ISSUE #77: feat: historical period comparison on stat cards
## Summary
Compare the current date range against a previous period of equal length, showing delta arrows and percentage changes on stat cards.

## Behavior
- When a date range is active, compute a "previous period" of equal length ending where the current range starts
- Example: if viewing Mar 1–31, compare against Feb 1–28
- In StatsGrid, show the comparison value and a delta indicator (▲/▼ + percentage)
- Color the delta: **green** for improvements (lower RHR, faster pace), **red** for regressions
- Only show deltas when the previous period has **>3 sessions** (avoid misleading comparisons)

## Metrics to Compare
- Avg session duration
- Avg HR
- Avg pace (run/walk tabs)
- Total distance
- Resting HR (if available in both periods)

## Implementation Details
- Create `lib/periodComparison.ts` — takes current and previous period data, returns delta objects
- Modify `StatsGrid` component to accept optional comparison data and render deltas
- Compute previous period in `Dashboard.tsx` using the existing `applyDateFilter` utility
- Delta display: small text below the main stat value, e.g. "▼ 3.2%" in green

## Test Plan
- Unit test `computePeriodDeltas()` with known current/previous data
- Test edge cases: no previous data, insufficient sessions, identical periods
- Verify delta arrows render correctly and colors match improvement direction
- Screenshot the stat cards with deltas using Playwright

## Files to Create/Modify
- `src/lib/periodComparison.ts` (new)
- `src/lib/periodComparison.test.ts` (new)
- `src/components/StatsGrid.tsx` (modify)
- `src/components/Dashboard.tsx` (compute previous period)

### PR DESCRIPTION
**Pull request by lucasklawrence**
## Summary

- Adds a five-card stats row above the Stair charts: **Sessions**, **Total time**, **Avg duration**, **Avg HR**, **Avg max HR**.
- Each card compares the active `DateFilter` range against a **previous period of equal calendar length, ending the day before `range.start`** — e.g. a current `Mar 1 → Mar 31` window compares against `Jan 29 → Feb 28`.
- Deltas are **suppressed when the previous period has fewer than 4 sessions** (issue's `>3` rule) to avoid small-sample swings advertising a 100%-style change.
- HR metrics render with **neutral coloring** because "improvement" depends on which zone you were targeting; sessions / total time / avg duration use the higher-is-better green/red palette.

Lifts the range + delta math into `lib/training-facility/period-comparison.ts` and the renderer into a generic, metric-spec-driven `CardioStatsCards` so Treadmill (#68) and Track (#69) can pick it up later with their own metric arrays.

## Test plan

- [x] `npm test` — vitest unit suite, 355/355 passing (22 new for this feature)
- [x] `npx tsc --noEmit` — clean
- [x] `npx next build` — clean
- [x] `npm run test:e2e` — Playwright 14/14 (no e2e specs touch the Stair view, but ran for safety)
- [ ] Open `/training-facility/gym/stair` on the Vercel preview, confirm the card row renders between the date filter and the chart grid
- [ ] Switch presets (1M / 3M / 6M / 1Y / All) — current values update, deltas update or collapse to "No prior comparison" when the previous window has ≤3 sessions
- [ ] Edit the custom date inputs — same behavior
- [ ] Mobile width (390 px): cards wrap to 2 cols, then 1
- [ ] No regressions on the existing chart grid or session log table below

Closes #77.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: lib/training-facility/period-comparison.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+ *   spans the same duration ending the day before `range.start`.
+ */
+export function computePreviousRange(range: DateRange): DateRange {
+  const durationMs = range.end.getTime() - range.start.getTime()
+  // End of previous range = the day before `range.start`. Snap to
+  // end-of-day so downstream filters that compare against
+  // `parseSessionDate(s.date).getTime()` (local midnight) include
+  // every session on that day.
+  const previousEnd = endOfDay(new Date(range.start.getTime() - 1))
+  const previousStart = startOfDay(new Date(previousEnd.getTime() - durationMs))
```
[the remark this answers is not in the document]
Comment (lucasklawrence) [reply lib/training-facility/period-comparison.ts]: Good catch — fixed in dd817b4. Switched the previous-range derivation to local-date `setDate` arithmetic and the day-count helper to UTC-midnight subtraction so DST transitions don't drift the result by ±1 day. Added regression tests covering both fall-back (Nov 1 → Nov 30 = 30 days) and spring-forward (Mar 1 → Mar 31 = 31 days) windows.

**Comment by lucasklawrence** [comment]
## Ready to merge ✅

| Check | Status |
| --- | --- |
| Playwright `e2e` | ✅ pass (1m27s) |
| Vercel deploy | ✅ pass |
| CodeRabbit | ✅ pass (no findings) |
| Codex | ✅ DST P1 finding addressed in dd817b4 with regression tests, [thread resolved](https://github.com/lucasklawrence/courtfolio/pull/141#discussion_r3158717628) |
| Branch vs `main` | 2 ahead / 0 behind, `mergeStateStatus: CLEAN` |

**Preview:** https://lucas-home-court-git-feat-77-pe-0a382e-lucasklawrences-projects.vercel.app/training-facility/gym/stair (gated — same render is on local dev `http://127.0.0.1:3010/training-facility/gym/stair`)

### Screenshots — `/training-facility/gym/stair` with real data (539 sessions imported)

**Desktop — 1280×800**
desktop — stair detail view

**Mobile — 390×844**
mobile — stair detail view

The mobile shot includes the Next.js dev indicator in the corner — that's a dev-only artifact, not in production.

### Test plan

- [x] `npm test` — 357/357
- [x] `npx tsc --noEmit` — clean
- [x] `npx next build` — clean
- [x] `npm run test:e2e` — 14/14
- [x] Local preview spot-check at 1M / 1Y presets with real Apple Health data
- [x] DST regression test added (Nov 1 → Nov 30 = 30 days, Mar 1 → Mar 31 = 31 days)
- [ ] One last preset cycle on the preview before merge if you want it

### Merge

\`\`\`
gh pr merge 141 --squash --delete-branch
\`\`\`

After merge, the screenshots branch can be deleted (cleanup): \`git push origin --delete screenshots-pr-141\`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)
